numpy/core/src/multiarray/common.c

```diff
@@ -214,6 +213,13 @@ PyArray_DTypeFromObjectHelper(PyObject *obj, int maxdims,
             int itemsize;
             PyObject *temp;
 
+            /*
+             * For this branch, _array_find_python_scalar is only good for
+             * finding that it is a scalar.
+             */
```

Comment:
I don't really understand this comment. Is the point you're making that the input dtype is completely ignored?

Reply:
Yeah, that is better.